actions/poll_status.py

```diff
@@ -25,17 +29,50 @@ def query(self, url, verify_ssl_cert):
             resp = requests.get(url, verify=verify_ssl_cert)
             return resp
         except RequestException as err:
-            current_time = datetime.datetime.now()
-            self.logger.warning(
-                "{0} -- {1} - an error was encountered: {2}".format(current_time, url, err))
-            return None, None
+            self.logger.warning("An error was encountered when "
+                                "querying url: {0},  {1}".format(url, err))
+            return None
+
+    def post_to_endpoint(self, endpoint, body, irma_mode, verify_ssl_cert):
 
-    def run(self, url, sleep, ignore_result, verify_ssl_cert, max_retries=3):
+        def _rewrite_link(link):
+            endpoint_parsed = urlparse(endpoint)
+            first_part_of_path = endpoint_parsed.path.split('/')[1]
+            link_parsed = urlparse(link)
+            return "{}://{}/{}{}?{}".format(endpoint_parsed.scheme,
+                                            endpoint_parsed.netloc,
+                                            first_part_of_path,
+                                            link_parsed.path,
+                                            endpoint_parsed.query)
```

Comment:
At the moment we are just passing the `apikey` as an argument on the URL, but I guess that could change someday, somehow. Hopefully the destination microservice should behave nicely even with unexpected queries arrive, but you never know.